Move details about refund policies for cart items to `CheckoutTerms`

fredrikekelund opened the pull request:
#### Proposed Changes

🚨 This description has been updated after several rounds of reviews and fixes.

A while back, in #63489, we added more detailed info to the checkout page about refund windows for different types of cart items. That change sparked a discussion about whether this was really the best way to display those details and we [concluded](https://github.com/Automattic/wp-calypso/pull/63489#issuecomment-1144957120) that no, `CheckoutTerms` (i.e. just above the `Pay now` button) would be a better placement.

This PR reduces the information in `WPCheckoutOrderSummary` (i.e. the right sidebar) to a single row and adds a new `RefundPolicies` component that replaces `DomainRefundPolicy` in the checkout terms. `RefundPolicies` displays information about refund windows for all refundable product types.

I have also incorporated a request from Nomado to change how domain refunds are communicated in the checkout summary. The changes here will probably be followed up by another PR that adds some more text to the checkout summary for cases when all checkout items are domains.

In `WPCheckoutOrderSummary`, the following logic applies:

1. If the cart contains no refundable products, display nothing
2. If the cart contains only domain registrations/renewals, display nothing
3. If the cart contains any non-refundable products, display nothing
4. If there is a single refund window in the cart AND there is a plan+free domain bundle, display `%(days)d-day money back guarantee for %(planName)s`
5. If there is a single refund window in the cart and no plan+domain bundle, display `%(days)d-day money back guarantee`
6. If there is a monthly plan+paid domain "bundle" in the cart, display `%(days)d-day money back guarantee for %(planName)s`
7. If the cart contains only a plan renewal and a domain renewal, display `%(days)d-day money back guarantee for %(planName)s`
8. If there are multiple refund windows in the cart, display `%(minimumRefundWindow)d-day full money back guarantee`

| Products | Screenshot |
| - | - |
| Yearly plan, bundled domain | ![01-yealy-plan-domain-bundle](https://user-images.githubusercontent.com/1101677/194013969-2c94be3d-aa89-4c43-8489-7614cd4a79e6.png) |
| Yearly plan, bundled domain, Professional Email trial | ![02-yearly-plan-domain-bundle-professional-email](https://user-images.githubusercontent.com/1101677/194013977-f61d627b-6e67-47c9-b641-98be1a16126e.png) |
| Yearly plan, bundled domain, add-on | ![03-yearly-plan-domain-bundle-add-on](https://user-images.githubusercontent.com/1101677/194013980-2858aac5-e6ed-4014-8d0d-fb7a622117e4.png) |
| Monthly plan, "bundled" domain, add-on | ![04-monthly-plan-domain-bundle-add-on](https://user-images.githubusercontent.com/1101677/194013983-5c8466c2-df82-47dc-b8e5-9d7c4aa1cad9.png) |
| Monthly plan, "bundled" domain | ![05-montly-plan-domain-bundle](https://user-images.githubusercontent.com/1101677/194013987-6bf44cb8-0d90-48e8-8a62-599f4d872717.png) |
| DIFM Lite | ![difm-after](https://user-images.githubusercontent.com/1101677/191489013-84987807-ddc6-42bc-8839-2bfc1242b6ba.png) |


#### Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

We want to test with a few different types of products, so bear with me.

1. Go to `/start/onboarding-with-email`
2. Add a domain, a mailbox and a plan and proceed to checkout
3. Ensure that a `14-day money back guarantee for {{ planName }}` row is visible in the right sidebar
4. Ensure that the following text row is visible in `CheckoutTerms`: `You understand that domain name refunds are limited to 96 hours after registration and yearly plan refunds are limited to 14 days after purchase. Refunds of paid plans will deduct the standard cost of any domain name registered within a plan.`
5. Complete checkout
6. Navigate to `/setup?siteSlug=<site_slug>`
7. Check the `Hire a professional to design my website` checkbox and proceed all the way to checkout
8. Ensure that **no** `Money back guarantee` row is visible in the right sidebar
9. Ensure that **no** information about refunds is visible in `CheckoutTerms`
10. Empty the cart
11. Set up the mailbox you bought earlier and then click `Renew now` on the email management page
12. Ensure that a `14-day money back guarantee` row is visible in the right sidebar 
13. Ensure that the following text row is visible in `CheckoutTerms`: `You understand that refunds are limited to 14 days after purchase or renewal for non-domain products with yearly subscriptions.`
14. Empty the cart
15. Add a new domain to your cart (`/domains/add/:site_slug`) and proceed to checkout
16. Ensure that **no** `Money back guarantee` row is visible in the right sidebar
17. Ensure that **no** information about refunds is visible in `CheckoutTerms`

#### Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #

fredrikekelund committed:
Move details about refund policies for cart items from WPCheckoutOrderSummary to CheckoutTerms

fredrikekelund committed:
Added unit tests

fredrikekelund committed:
Remove some RefundPolicy types

fredrikekelund committed:
Removed more RefundPolicy types

olaseni commented on client/my-sites/checkout/composite-checkout/components/refund-policies.tsx:
```diff
@@ -0,0 +1,139 @@
+			onClick={ () => gaRecordEvent( 'Upgrades', 'Clicked Refund Support Link' ) }
+		/>
+	);
+	const cancelDomainSupportPage = (
```
This copy is different from the original with the word 'Link'

fredrikekelund replied:
Keen eyes! 👍

olaseni reviewed (approved):
This looks good IMO, but I'll ping folks from @Automattic/payments  to take a look as well as the checkout section isn't my forte.

<img width="948" alt="Screenshot 2022-09-22 at 2 04 40 PM" src="https://user-images.githubusercontent.com/277661/191767858-99fde52e-9c2f-4868-8554-38fcc15f4e83.png">
<img width="966" alt="Screenshot 2022-09-22 at 2 02 55 PM" src="https://user-images.githubusercontent.com/277661/191767870-29c4efc7-18af-454f-b6ac-7e1e90f0e090.png">

DavidRothstein commented:
"Money back guarantee" on its own (with no time period) seems very vague to me.  As a customer, I would personally be distracted by this and wonder if it's too good to be true or there's a catch somewhere.  (Yes, the details are further down, but people won't tend to look at those.)

If the goal is to reduce the amount of text from what's there now, could you consider having a single "X day money-back guarantee" line instead (where X is the shortest refund window among all the products in the cart)?  Or maybe modify that for the common case of a plan + domain to mention both refund windows in one line in that specific case... but even then it would be a lot simpler than the current text.

Also, a couple issues I found while testing this.

**1. All of the new text is missing for two-year plans or one-time purchases.**

Two-year plan:

![two-year](https://user-images.githubusercontent.com/235183/191881030-9402c579-b0aa-49d1-b34b-e7614064e612.png)

One-time purchase (premium theme):

![premium-themes](https://user-images.githubusercontent.com/235183/191881421-568eee86-2aac-4297-94c2-c73e4ccaf480.png)

**2. When renewing a plan and domain at the same time, the combination of the two bullet points makes it sound like the refund window for the domain subscription will be 14 days rather than 96 hours (since the first bullet point talks about renewals and the second doesn't):**

![renew-plan-and-domain](https://user-images.githubusercontent.com/235183/191881161-b4ec55b5-ad33-4af2-afc9-19b9a53b4a15.png)

fredrikekelund committed:
Maybe show refund window in CheckoutSummaryFeaturesList

fredrikekelund committed:
Support paid themes and biennial products

fredrikekelund committed:
Adapted message in CheckoutSummaryFeaturesList

fredrikekelund committed:
Simplified message in CheckoutSummaryFeaturesList

fredrikekelund committed:
Specific copy regarding plan refunds

fredrikekelund committed:
Reduce domain related refund windows

fredrikekelund committed:
More precise reduction of domain related refund windows

fredrikekelund commented:
Thanks for the review @DavidRothstein!

First, regarding your points **1 and 2**.

1. I have addressed the problem with biennial products and paid premium themes.
2. I feel like this could be addressed by improving the copy. I had a stab at this by bringing back dedicated `RefundPolicy`s for the different plan terms (as opposed to using `GenericMonthly`/`GenericYearly`). See screenshots below for result.

> "Money back guarantee" on its own (with no time period) seems very vague to me.

This string was previously the fallback for when we couldn't determine a single refund window for the products in the cart. I agree that it's quite vague though, so this would be a good time to reevaluate it!

I tried a new implementation based on your suggestions. Here's a summary: 
1. Revert to displaying `(n)-day money back guarantee` if there is just one refund policy in the cart. 
2. Prioritise plans in a way so that users who are purchasing a plan with a bundled domain see `(n)-day money back guarantee for plan` in the sidebar.
3. For more complex scenario, we fall back to the string `Limited money back guarantee`. I thought this was better than using the lowest refund window, because that can be misleading in cases with complex combinations of products.

Here are updated screenshots that illustrate what the user sees with different combinations of products.

| Products | Before | After |
| - | - | - |
| Paid theme | ![theme-before](https://user-images.githubusercontent.com/1101677/191949253-e069fbac-cdcd-4093-aa8b-cebc4deca556.png) |  ![theme-after](https://user-images.githubusercontent.com/1101677/191949267-e809c255-0363-4246-9c86-5d66302995a6.png) |
| Plan, bundled domain and free professional email trial | ![plan-with-email-before](https://user-images.githubusercontent.com/1101677/191949330-9c4c3df0-8700-4c11-8014-eeab9a517af8.png) | ![plan-with-email-after](https://user-images.githubusercontent.com/1101677/191949347-3107275d-eefb-46ae-91ac-cf0f21603d4a.png) |
| Plan, bundled domain and paid domain | ![weird-before](https://user-images.githubusercontent.com/1101677/191949411-cbeecbc7-2a27-413f-b27c-d2c4aa7b1700.png) | ![weird-after](https://user-images.githubusercontent.com/1101677/191949432-5bbc59cc-9f7e-410b-a1f8-23e4b7ce6ba4.png) |

fredrikekelund committed:
Typo fix

DavidRothstein commented:
Thanks -- overall it looks a lot better.

>This string was previously the fallback for when we couldn't determine a single refund window for the products in the cart. I agree that it's quite vague though, so this would be a good time to reevaluate it!

I did see that "Money back guarantee" was already in the code, but as far as I can tell, it actually only shows up currently when the refund window is 0 days for a _particular_ product?  Based on how the code is called here:

https://github.com/Automattic/wp-calypso/blob/150e0f8fa838dbdef556a34f9d44fb65dc13be96/client/my-sites/checkout/composite-checkout/components/wp-checkout-order-summary.tsx#L225

and then how the fallback gets used here:

https://github.com/Automattic/wp-calypso/blob/150e0f8fa838dbdef556a34f9d44fb65dc13be96/client/my-sites/checkout/composite-checkout/lib/get-refund-text.ts#L17

In other words, it seems to be displayed for non-refundable products, which exactly when there is **no** money-back guarantee at all :frowning_face:  So definitely worth revising all this!

>Prioritise plans in a way so that users who are purchasing a plan with a bundled domain see `(n)-day money back guarantee for plan` in the sidebar.

Is that clear enough, though?  The plan is the only thing they're actually paying money for, so based on that, I would expect to get **all** my money back if cancelling within `n` (i.e. 14) days.  However, if they cancel after 4 days, they won't actually get all their money back.

>For more complex scenarios, we fall back to the string `Limited money back guarantee`. I thought this was better than using the lowest refund window, because that can be misleading in cases with complex combinations of products.

I see what you're saying, just not sure "Limited" is totally clear.  For what it's worth, the reason I suggested using the lowest refund window is that it actually does represent the money-back guarantee period for the **total** price they are paying -- and this text is shown right above the total price in their shopping cart.  So if they want to get all their money back, that's the refund window that matters (even if it's also true that they can get some of their money back beyond that).

FYI, I saw this when purchasing a yearly plan and add-on at the same time (it's not wrong, but the text on the left is repetitive and the text on the right definitely could have said "14 days"):

![plan-plus-add-on](https://user-images.githubusercontent.com/235183/192192450-c0a685a2-eb18-42b6-94de-216d08a06c85.png)

fredrikekelund committed:
Further reduce refund policies in checkout summary

fredrikekelund committed:
Use domain refund window in summary for plan+domain

fredrikekelund committed:
Remove use of isTheme function

fredrikekelund committed:
Refactor plan+domain bundle refund policy messaging

fredrikekelund committed:
Never use "Limited money back guarantee"

fredrikekelund commented:
> The plan is the only thing they're actually paying money for, so based on that, I would expect to get all my money back if cancelling within n (i.e. 14) days. However, if they cancel after 4 days, they won't actually get all their money back.

Fair point. I've changed it so that when there are multiple refund windows in the cart, we display `{{ shortestRefundWindow }}-day _full_ money back guarantee`.

If we want a single row in `WPCheckoutOrderSummary`, this seems like a pretty well-measured approach. It could work to display one row about the domain and another one about the plan, but that might start to conflate the checkout summary with the checkout terms again.

> the text on the left is repetitive

Plans without a bundled domain and add-ons now share the same refund policy again, which should be an improvement in your test case.

I've also tweaked the messaging for plan+domain bundles into a single row that says e.g. `You understand that domain name refunds are limited to 96 hours after registration and yearly plan refunds are limited to 14 days after purchase. Refunds of paid plans will deduct the standard cost of any domain name registered within a plan.`

We can tweak that string, but it seems beneficial to talk about the domain and the plan in a single row, given how intertwined they are when they are sold as a bundle.

> the text on the right definitely could have said "14 days"

Good point! I've updated the logic here.

Below are screenshots from the current state of things:

| Products | Before | After |
| - | - | - |
| Personal plan and add-on | ![plan-addon-before](https://user-images.githubusercontent.com/1101677/192244793-44cec8de-fa2d-46f9-860a-67554703c317.png) | ![plan-addon-after](https://user-images.githubusercontent.com/1101677/192244810-1b6760ab-8fac-489d-b4d0-79d2c2404748.png) |
| Plan, bundled domain and free professional email trial | ![plan-email-before](https://user-images.githubusercontent.com/1101677/192245117-d13c6ccc-5743-4099-90c9-9d83489bda08.png) | ![plan-email-after](https://user-images.githubusercontent.com/1101677/192245134-aef24623-e034-4b4d-863b-1585d986f1dc.png) |
| Plan, bundled domain and paid domain | ![weird-before](https://user-images.githubusercontent.com/1101677/192244926-3b9e1466-cc75-4e86-a073-56218c7c19ed.png) | ![weird-after](https://user-images.githubusercontent.com/1101677/192244945-2e1a49f9-a121-48a3-97f6-ffae7cadd7f6.png) |

fredrikekelund committed:
Simplify getRefundPolicies

fredrikekelund committed:
Comment with clarification

delputnam commented:
Regarding the "4-day Money back guarantee" message, since that's specific to domains and the MBG policy for domains is under review at the moment, we (Nomado) wonder if it might be better to remove the reference to this in the sidebar completely.

fredrikekelund committed:
Reverting checkout summary text for plan and domain bundles

fredrikekelund committed:
Update checkout summary

fredrikekelund commented:
After discussing with Nomado, I opted to change back the checkout summary to display `n-day money back guarantee for plan` when the user has a plan+domain bundle in their cart.

fredrikekelund committed:
De-emphasize domain refunds entirely in checkout summary

DavidRothstein commented:
I still think "14 day money back guarantee for plan" is confusing here, because if these users cancel their purchase between 4-14 days, they won't actually get all the money back that they spent on the plan.

There are a number of other ways we could communicate this.  My suggestion of "4 day money-back guarantee" is still valid (we could remove that sentences from domain _purchases_ as requested, without having to remove it from plan purchases that also include a free domain).  Or else use something like **"14 day money-back guarantee (minus domain registration costs)"**?

That said, I do also see that this is effectively a preexisting issue.  So if we don't want to address it here, it's OK I suppose, but in that case can we preserve the exact wording we have already?  In other words, instead of seeing this when purchasing a plan and free domain:

![purchase-after](https://user-images.githubusercontent.com/235183/192884126-ba5de675-572a-494b-9bbd-77770829bfbb.png)

Just keep it like this (the current behavior):

![purchase-before](https://user-images.githubusercontent.com/235183/192884608-4e5bdb22-be25-4f29-80e6-6bc5718f5145.png)

It's much more clear to continue mentioning the specific plan name (and we should be as clear as possible if other aspects of it are potentially misleading).

---

As for renewals, I'm seeing the following when trying to renew a plan and domain (using the latest version of the pull request):

![renew-after](https://user-images.githubusercontent.com/235183/192884565-b5762e98-2785-4429-860d-18cf95a3b1ae.png)

That looks wrong -- I think the "for [plan]" wording is needed in this case also, since the 14 days doesn't apply to everything that is being renewed.

fredrikekelund committed:
Adjustments to checkout summary

fredrikekelund commented:
> 14 day money-back guarantee (minus domain registration costs)

Initially I thought this was quite verbose, but I came around to liking it. It's accurate and hopefully accommodates the conflicting desires here.

> As for renewals, I'm seeing the following when trying to renew a plan and domain.

I made a number of changes to the checkout summary to try and account for all the different scenarios. I wasn't able to add a plan and a domain renewal to my cart at the same time, but I'd appreciate if you could take another look @DavidRothstein.

Here's another set of screenshots with the current state of the checkout summary. @delputnam if you could also take a quick look at these, that'd be great!

| Products | Screenshot |
| - | - |
| Monthly plan, domain, add-on | ![01 Monthly plan, domain, add-on](https://user-images.githubusercontent.com/1101677/193046480-b0ce16ef-cea0-4499-be25-aca98de49eda.png) |
| Yearly plan, domain, add-on | ![02 Yearly plan, domain, add-on](https://user-images.githubusercontent.com/1101677/193046530-b78a0e34-0258-4591-b70c-41f63290fb17.png) |
| Yearly plan, domain | ![03 Yearly plan, domain](https://user-images.githubusercontent.com/1101677/193046590-3c8c5482-9ced-423c-93ee-ef1b55474a8f.png) |
| Paid domain | ![04 Domain](https://user-images.githubusercontent.com/1101677/193046659-145f13b0-c487-4173-b01d-92efe63c6dcc.png) |

sirbrillig reviewed (approved):
I know there's still some discussion around wording which I won't get involved in here, but from a checkout code perspective, this is a nice improvement! Thanks!

fredrikekelund committed:
Alphabetical sorting

fredrikekelund committed:
Firstly account for non-refundable products

fredrikekelund committed:
Added a case specifically for monthly plan+paid domain

fredrikekelund commented:
Summarizing the logic in the checkout summary as it currently stands. 

The product combinations from my latest set of screenshots will still generate the same text, but I have added a case specifically for monthly plans+paid domains (case 6).

1. If the cart contains no refundable products, display nothing
2. If the cart contains only domain registrations/renewals, display nothing
3. If the cart contains any non-refundable products, display `Partial money back guarantee`
4. If there is a single refund window in the cart AND there is a plan+free domain bundle, display `%(days)d-day money back guarantee (minus domain registration costs)`
5. If there is a single refund window in the cart and no plan+domain bundle, display `%(days)d-day money back guarantee`
6. If there is a monthly plan+paid domain "bundle" in the cart, display `%(days)d-day money back guarantee for %(planName)s`
7. If there are multiple refund windows in the cart, display `Minimum %(days)d-day money back guarantee`

My guess is that the majority of checkouts will be covered by cases 2, 4 or 5.
6 is probably also not an uncommon scenario, which is why I added a case specifically for that.
3 seems like a fairly uncommon scenario, but I find it important to set user assumptions correctly for that case.

DavidRothstein commented:
It looks mostly good to me, thanks!

Personally, I would say that phrases like "Partial money back guarantee" and "Minimum X-day money back guarantee" are muddying the waters, and distracting the user by having them try to figure out what it means... For the first, I would just remove the sentence entirely (don't say anything about a money back guarantee if some products in the cart are non-refundable), and for the second, I'd still consider just removing "Minimum", but you could also go with "X-day full money back guarantee" (like you had at one point before)?

To be honest, my "minus domain registration costs" suggestion could potentially have a similar problem (of making the user think too much)...

Who should do a final review of all the wording changes here?  (I think these kinds of changes can typically have a real impact on sales, in one direction or the other.)

Regarding renewals, I tested a plan and domain renewal by putting one in the shopping cart, backing out, and then putting the other in the shopping cart (after previously buying them using PCYsg-IA-p2) -- a real user would typically use a link in a billing email to renew both at the same time, but for testing purposes my way should be fine.  Results look correct to me:

![plan-and-domain-renewal](https://user-images.githubusercontent.com/235183/193358165-dd20f89f-d56c-456b-831f-effd733db9c1.png)

One more thing: Looking at the above screenshot, would it be better if the third informational bullet point on the left said "You understand that refunds are limited to 14 days after purchase or renewal for **non-domain** products with yearly subscriptions" instead (emphasis added)?  Without that it almost seems like the third and fourth bullet points contradict each other (given that domain renewals are billed yearly).

fredrikekelund committed:
Minor changes to checkout summary

- If there are any non-refundable products in cart, display nothing (instead of "Partial money back guarantee")
- Revert checkout summary string for plan bundles to "n-day money back guarantee for WordPress.com Premium"

fredrikekelund committed:
Include "non-domain" in generic refund policy strings

fredrikekelund committed:
Plan+domain renewal case in checkout summary

fredrikekelund commented:
Given that we've increased the scope here with Nomado's request, my preference would be to reduce the overall impact of this PR in order to get it shipped. This should hopefully also help isolate the potential impact on conversions by not tweaking the knobs where we don't have to.

With that goal in mind, I took up these suggested changes from you @DavidRothstein:

- If there are any non-refundable products in cart, display nothing (instead of `Partial money back guarantee`)
- Revert checkout summary string for plan bundles to `n-day money back guarantee for {{ planName }}`
- Include `[...] non-domain [...]` in the generic refund policy details

Regarding renewals, it seems I just somehow misunderstood the test case, thinking that one product was supposed to be a renewal and the other one was not 🤷‍♂️ In any case, to fully accommodate Nomado, I made an adjustment specifically for plan+domain renewals so that this combination now also displays `n-day money back guarantee for {{ planName }}`.

> Who should do a final review of all the wording changes here?

I pinged about this here 

For easier overview, here's another summary of the logic in the checkout summary:

1. If the cart contains no refundable products, display nothing
2. If the cart contains only domain registrations/renewals, display nothing
3. If the cart contains any non-refundable products, display nothing
4. If there is a single refund window in the cart AND there is a plan+free domain bundle, display `%(days)d-day money back guarantee for %(planName)s`
5. If there is a single refund window in the cart and no plan+domain bundle, display `%(days)d-day money back guarantee`
6. If there is a monthly plan+paid domain "bundle" in the cart, display `%(days)d-day money back guarantee for %(planName)s`
7. If the cart contains only a plan renewal and a domain renewal, display `%(days)d-day money back guarantee for %(planName)s`
8. If there are multiple refund windows in the cart, display `%(days)d-day full money back guarantee`

DavidRothstein commented:
That logic seems pretty solid to me now (although I didn't test most of it).

I did test the renewal case again, and it looks to me like the latest version is displaying some text in the Terms that shouldn't be there?

I'm seeing this for a yearly plan renewal, which looks wrong:

![plan-renewal](https://user-images.githubusercontent.com/235183/193608019-aad62ee8-a73d-4554-b48a-ccb93e00f161.png)

And this for a yearly plan+domain renewal (which also looks wrong -- seems to be showing the new purchase text for domains, with the regular renewal text for domains in the bullet point underneath it):

![plan-and-domain-renewal](https://user-images.githubusercontent.com/235183/193608015-adce257a-fee2-43f0-907e-05b2b3766350.png)

Also noticed that when using a free domain credit (after previously buying a plan), one of the lines in the Terms is no longer there.  This is the "before" screenshot showing the line that went missing:

![free-domain-credit](https://user-images.githubusercontent.com/235183/193611979-57099d58-ee27-4af6-85d3-f2838666952c.png)

I suspect removing it was intentional, but I do wonder if including it originally was supposed to be some kind of attempt to communicate the "...Refunds of paid plans will deduct the standard cost of any domain name registered within a plan" idea that is shown in other scenarios.  (These users might otherwise see no message anywhere that is relevant to how their use of the free domain credit affects a potential refund of their plan later on.)

fredrikekelund commented:
Thanks for your keen eyes, @DavidRothstein! I've addressed the issue you pointed out where plan/domain renewals would display the wrong text in the checkout terms.

As for the refund terms for bundled domains purchased separately from a paid plan, I've added a specific case for this that shows the following text (a string that's already used on the checkout page today):

> You understand that {{refundsSupportPage}}domain name refunds{{/refundsSupportPage}} are limited to 96 hours after registration. Refunds of paid plans will deduct the standard cost of any domain name registered within a plan.

This policy and the `DomainNameRegistration` refund policy are mutually exclusive. Here's a screenshot from the checkout page with one bundled domain and one paid domain in the cart:

![bundled-domain-paid-domain](https://user-images.githubusercontent.com/1101677/193765918-420ab453-f627-46cc-b00b-d835254553b5.png)

fredrikekelund committed:
Fix wrong terms displayed for plan renewals

fredrikekelund committed:
Added refund policy for bundled domain without paid plan

fredrikekelund committed:
Move refund policy deduplication to RefundPolicies

fredrikekelund committed:
Fix refund policy order in unit test

DavidRothstein commented:
Only had time for a quick look (and am AFK tomorrow) but it looks great to me -- thanks!

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7568760

Thank you @fredrikekelund for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7568760).

DavidRothstein commented:
I found a couple relatively minor problems here and created followups for them:
- https://github.com/Automattic/wp-calypso/issues/69142
- https://github.com/Automattic/payments-shilling/issues/1167

Also a followup for the (preexisting) issue regarding the implication that you can get all your money back within 14 days for a plan purchased with a free domain:
- https://github.com/Automattic/payments-shilling/issues/1168
